community/codegen/src/main/java/org/neo4j/codegen/source/ClassWriter.java

```diff
@@ -33,10 +33,51 @@
 class ClassWriter implements ClassEmitter
 {
     private final StringBuilder target;
+    final Configuration configuration;
 
-    ClassWriter( StringBuilder target )
+    ClassWriter( StringBuilder target, Configuration configuration )
     {
         this.target = target;
+        this.configuration = configuration;
+    }
+
+    void declarePackage( TypeReference type )
+    {
+        append( "package " ).append( type.packageName() ).append( ";\n" );
+    }
+
+    void javadoc( String javadoc )
```

Comment:
We already use it, the generated source code contains a comment saying "this is generates code".

We had that before this refactoring, this refactoring just makes it more clear what it is.